engine/src/main/java/org/camunda/bpm/engine/impl/cmd/ModifyProcessInstanceCmd.java

```diff
@@ -84,7 +84,9 @@ public Void execute(CommandContext commandContext) {
     processInstance = executionManager.findExecutionById(processInstanceId);
 
     if (!processInstance.hasChildren()) {
-      if (processInstance.getActivity() == null) {
+      if (processInstance.isCanceled() && processInstance.isRemoved()) {
+        processInstance.propagateEnd();
```

Comment:
If I remember correctly we don't need to propagate an end here, I will double check
```suggestion

```